### PR TITLE
Migrate to SignalFX error tags

### PR DESCRIPTION
**Pull request by owais**
Replace OpenTracing error logs with SignalFx tags

This PR changes how errors are recorded on spans. It uses the new
SignalFx semantic convention and records the errors as attributes
instead of logs.

In addition to updating errors, also fixed postgresql and elasticsearch
failing tests.

### CODE REVIEW

File: src/plugins/knex.js [hunk 1]
```diff
@@ -62,11 +62,12 @@ function createWrapRunnerQuery (tracer, config) {
...
-  span.log({
-    'error.type': error.name,
-    'error.msg': error.message,
-    'error.stack': error.stack
+  span.addTags({
+    'sfx.error': true,
+    'sfx.error.kind': error.name,
+    'sfx.error.object': error.name,
+    'sfx.error.message': error.message,
+    'sfx.error.stack': error.stack
```
Comment (owais) [inline src/plugins/knex.js]: Turns out knex integration actually fails to capture errors right now. Updated code to use SignalFx error tags for now. Will address the actual error handling issue in an upcoming PR.

File: src/format.js [hunk 2]
```diff
@@ -85,9 +85,10 @@ function extractError (trace, span) {
   const error = span.context()._tags['error']
 
   if (error instanceof Error) {
-    trace.meta['error.msg'] = error.message
-    trace.meta['error.type'] = error.name
-    trace.meta['error.stack'] = error.stack
+    trace.meta['sfx.error.message'] = error.message
+    trace.meta['sfx.error.object'] = error.name
+    trace.meta['sfx.error.kind'] = error.name
```
Comment (rmfitzpatrick) [inline src/format.js]: Is the redundant kind tag necessary?  I see `error.kind` is considered optional in OT, though error.object should be the instance itself (`error.toString()`)?

Comment (owais) [reply src/format.js]: I'm not entirely sure yet so I put both here for now until I figure it out. `error.toString()` at first seems to be just concatination of `kind` and `message` and I'm not sure if that's right. In Java at least, `kind` seems to be the error/exception type name while as object is the full qualified path but I'm not 100% sure about this yet either. Need to look deeper into this and figure out what to use exactly.

Comment (johnbley) [reply src/format.js]: error.object is supposed to be the type of the exception/error object.  In Java that's like java.lang.IllegalArgumentException.  error.kind is supposed to be a shorter version of this (if possible - in the Java example it would be just IllegalArgumentException).    For node.js I think these will mostly be Error anyway so it's find to have them identical.

File: test/plugins/nest.spec.js [hunk 3]
```diff
@@ -121,14 +139,58 @@ describe('Plugin', () => {
...
-              axios
-                .get(`http://localhost:${port}/users`)
-                .catch(done)
-            })
+          axios
+            .get(`http://localhost:${port}/users`)
             .catch(done)
         }).timeout(5000)
+
+        it('should properly record errors', done => {
```
Comment (owais) [inline test/plugins/nest.spec.js]: This does not yet test for errors but modifies the test case in a way that allows multiple unit tests for nest. Earlier one server was being started per test. The tests for error handling will be added in a different PR.

**Comment by owais** [comment]
Cassandra and Koa are failing in master due to newer versions being released that are incompatible with the instrumentation. We'll need to update the instrumentation code to support the newer versions.

File: src/format.js (hunk 2 above)
Comment (rmfitzpatrick) [reply src/format.js]: @johnbley modeling [from these](https://opentracing.io/specification/conventions/) (Captured errors section) the error.object should be the actual instance, whose utility seems pretty limited to certain platforms.  I think what you've done in java makes sense there, but maybe not in node?

I may have lost the thread but was it decided the redundant kind is necessary?

**Comment by rmfitzpatrick** [comment]
@owais, would you be to update the supported version ranges with your discoveries?

**Comment by owais** [comment]
@rmfitzpatrick Updated `koa` to explicitly support only `<8` and submitted a PR to update cassandra instrumentation here: https://github.com/signalfx/signalfx-nodejs-tracing/pull/51

File: src/format.js (hunk 2 above)
Comment (owais) [reply src/format.js]: As far as I understand, `kind` doesn't seem redundant to me. `kind` is required and should always be there. `object` may or may not depending on the platform. Node does not have something that translates to "object" as in Java or Python. I'll omit object.

**Comment by owais** [comment]
@rmfitzpatrick rebases and updated koa. All tests are green now.
